### ISSUE TITLE
Use of exists: with erroneous code shows incorrect error in page template

### ISSUE THREAD

**Issue by davidjb** [malthe/chameleon#129]
By including this:

```
    <tal:block condition="exists:form">
    ...
    </tal:block>
```

in a page template, it appears that any error that occurs on the page, such as by including this problematic snippet:

```
${''.join([1,2,3])}
```

produces an incorrect traceback:

```
File "/home/david/.buildout/eggs/Chameleon-2.10-py2.7.egg/chameleon/template.py", line 125, in __call__
    return self.render(**kwargs)
  File "/home/david/.buildout/eggs/Chameleon-2.10-py2.7.egg/chameleon/zpt/template.py", line 257, in render
    return super(PageTemplate, self).render(**vars)
  File "/home/david/.buildout/eggs/Chameleon-2.10-py2.7.egg/chameleon/template.py", line 190, in render
    raise_with_traceback(exc, tb)
  File "/home/david/.buildout/eggs/Chameleon-2.10-py2.7.egg/chameleon/template.py", line 172, in render
    self._render(stream, econtext, rcontext)
  File "/tmp/tmpla4Pzm/profile_0a115cc2a1476969f981b6e5a3415492d4e5f193.py", line 2656, in render
    __macro.include(__stream, econtext.copy(), rcontext, __i18n_domain)
  File "/home/david/.buildout/eggs/Chameleon-2.10-py2.7.egg/chameleon/zpt/template.py", line 261, in include
    self._render(*args, **kwargs)
  File "/tmp/tmpNoXbaT/master_0dcd06fbeaf411b3407adc5bb4a14f8088378967.py", line 726, in render
    __slot_content(__stream, econtext.copy(), rcontext)
  File "/tmp/tmpla4Pzm/profile_0a115cc2a1476969f981b6e5a3415492d4e5f193.py", line 641, in __fill_content
    __content_140487793025840 = _lookup_attr('', 'join')([1, 2, 3, ])
TypeError: form

 - Expression: "
      ${''.join([1,2,3])}
      "
 - Filename:   ... files/src/templates/profile.pt
 - Location:   (45:18)
 - Source:     </tal:block>
                           ^
 - Expression: "form"
 - Filename:   ... files/src/templates/profile.pt
 - Location:   (43:35)
 - Source:     <tal:block condition="exists:form">
                                            ^^^^
```

Previously the `Expression` was incorrect too (clearly referencing the exists: statement) but I can't seem to reproduce this yet.  As shown, the location, source and expression reference the `exists:`.

After removing the `exists:` tal:block, the correct error is output:

```
TypeError: sequence item 0: expected string, int found

 - Expression: "
      ${''.join([1,2,3])}
      "
 - Filename:   ... files/src/templates/profile.pt
 - Location:   (42:12)
 - Source:     </div>
                     ^
```

Tested in Chameleon 2.10 and Python 2.7 (with Pyramid 1.4a2).

**Comment by lieryan** [comment]
Still reproducible in 2.15 and 2.24.

**Comment by malthe** [comment]
I see. This should definitely be fixed.

**Comment by malthe** [comment]
I can't reproduce this. Please post a complete template that fails for you.

**Comment by lieryan** [comment]
@malthe tested with Chameleon==2.24:

```
from chameleon import PageTemplate
body = '''\
<?xml version="1.0"?>
<root>
    <one tal:condition="exists: var_does_not_exists" />
    <two tal:attributes="my_attr dict()['key-does-not-exist']" />
</root>
'''
template = PageTemplate(body)
template()

```

Actual Traceback:

1. A weird amalgamation of error message: "KeyError: var_does_not_exists" (i.e. the correct error name but incorrect error message).

```
Traceback (most recent call last):
  File "test.py", line 10, in <module>
    template()
  File "/home/lieryan/.virtualenvs/tmp-f42733d8354a529/local/lib/python2.7/site-packages/chameleon/template.py", line 124, in __call__
    return self.render(**kwargs)
  File "/home/lieryan/.virtualenvs/tmp-f42733d8354a529/local/lib/python2.7/site-packages/chameleon/zpt/template.py", line 261, in render
    return super(PageTemplate, self).render(**vars)
  File "/home/lieryan/.virtualenvs/tmp-f42733d8354a529/local/lib/python2.7/site-packages/chameleon/template.py", line 191, in render
    raise_with_traceback(exc, tb)
  File "/home/lieryan/.virtualenvs/tmp-f42733d8354a529/local/lib/python2.7/site-packages/chameleon/template.py", line 171, in render
    self._render(stream, econtext, rcontext)
  File "a56b06593ae9e730d71f6e0ba777ae57.py", line 133, in render
KeyError: var_does_not_exists

 - Expression: "dict()['key-does-not-exist']"
 - Filename:   <string>
 - Location:   (line 4: col 33)
 - Expression: " var_does_not_exists"
 - Filename:   <string>
 - Location:   (line 3: col 31)
 - Arguments:  repeat: {...} (0)

```

Expected:

1. the real error, which is "KeyError: 'key-does-not-exists'", should be in the traceback with the correct error message
2. the AttributeError should be ignored completely instead of showing after the traceback, since it's within "exists:" clause.

**Comment by malthe** [comment]
Fixed in cce4163.
